core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/codesystem/CodeSystemEntry.java

```diff
@@ -203,153 +245,150 @@ private CodeSystemEntry(final String oid, final String name, final String shortN
+		return additionalProperties;
+	}
 
 	@Override
 	public int hashCode() {
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + ((oid == null) ? 0 : oid.hashCode());
-		result = prime * result + ((shortName == null) ? 0 : shortName.hashCode());
-		return result;
+		return Objects.hash(oid, shortName);
 	}
 
 	@Override
 	public boolean equals(final Object obj) {
-		if (this == obj)
-			return true;
-		if (obj == null)
-			return false;
-		if (!(obj instanceof CodeSystemEntry))
-			return false;
+		if (this == obj) { return true; }
+		if (obj == null) { return false; }
+		if (!(obj instanceof CodeSystemEntry)) { return false; }
+		
 		final CodeSystemEntry other = (CodeSystemEntry) obj;
-		if (oid == null) {
-			if (other.oid != null)
-				return false;
-		} else if (!oid.equals(other.oid))
-			return false;
-		if (shortName == null) {
-			if (other.shortName != null)
-				return false;
-		} else if (!shortName.equals(other.shortName))
-			return false;
-		return true;
-	}
-	
-	/**
-	 * Returns all code system short name dependencies and itself.
-	 */
-	@JsonIgnore
-	public SortedSet<String> getDependenciesAndSelf() {
-		ImmutableSortedSet.Builder<String> affectedCodeSystems = ImmutableSortedSet.naturalOrder();
-		affectedCodeSystems.addAll(getDependencies());
-		affectedCodeSystems.add(shortName);
-		return affectedCodeSystems.build();
-	}
-	
-	/**
-	 * Returns the short names of all affected code systems
-	 */
-	@JsonIgnore
-	public SortedSet<String> getDependencies() {
-		return TerminologyRegistry.INSTANCE.getTerminology(terminologyComponentId).getDependencies();
-	}
-	
-	/**
-	 * Returns a new branch path that originates from the code system's branch path
-	 */
-	@JsonIgnore
-	public String getRelativeBranchPath(String relativeTo) {
-		return String.format("%s%s%s", branchPath, Branch.SEPARATOR, relativeTo);
+
+		/*
+		 * FIXME: The original intention was to allow eg. local code systems to co-exist
+		 * in case of a short name collision, if they come from different sources (and
+		 * so their OID would be different), however:
+		 *
+		 * - The current implementation does not treat code systems with the same OID 
+		 *   as equal unless their short name also matches; 
+		 * 
+		 * - We can't change the implementation that returns true if _either_ the 
+		 *   short name _or_ the OID matches, as it goes against the transitive requirement 
+		 *   of the equivalence relation described in the javadoc of equals():
+		 * 
+		 *   code system | cs1       cs2       cs3
+		 *   short name  | ABC       CDE       CDE
+		 *   OID         | 1.2.3.4   1.2.3.4   5.6.7.8
+		 * 
+		 *   In this alternative world, cs1 = cs2 and cs2 = cs3, and so cs1 should be 
+		 *   equal to cs3, but it isn't!
+		 * 
+		 * - While some requests take care to look up both short name and OID for a code system,
+		 *   other parts (client and UI) practically treat short names as unique
+		 * 
+		 * Consider removing "oid" from the equality check.
```

Comment:
How about a few enforcing checks in the `CodeSystem` create API? Prevent creating new `CodeSystem` entries that use the same oid (or shortName) as another `CodeSystem`.
Or remove the possibility to get a `CodeSystem`s by its supposed to be unique oid. See the GET endpoint.

Reply:
Code system creation will fail if either of the identifiers is non-unique: https://github.com/b2ihealthcare/snow-owl/blob/37080602ee0c753cf419a761e78ff54af90fd77e/core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/codesystem/CodeSystemCreateRequest.java#L158-L165

The search request looks for the given identifier(s) in an either-or fashion, however: https://github.com/b2ihealthcare/snow-owl/blob/37080602ee0c753cf419a761e78ff54af90fd77e/core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/codesystem/CodeSystemSearchRequest.java#L46-L49